### PR TITLE
chore: Dedupe a whole bunch of dependency version specifications

### PR DESCRIPTION
**Pull request by Dinnerbone**
Anything that we specified 4 times or more...

### CODE REVIEW

File: core/build_playerglobal/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,6 @@ convert_case = "0.6.0"
 proc-macro2 = "1.0.79"
 quote = "1.0.35"
 swf = { path = "../../swf" }
-clap = {version = "4.5.4", features = ["derive"]}
+clap = {workspace = true, features = ["derive"]}
```
Comment (sleepycatcoding) [inline core/build_playerglobal/Cargo.toml:18]: Doesn't workspace declaration already set the ´derive` feature?

Comment (Dinnerbone) [reply core/build_playerglobal/Cargo.toml:18]: I guess it does!

**Comment by torokati44** [comment]
Partially addresses #15460.

Some more honorable mentions, for a future tweak PR:
```
downcast-rs
fnv
walkdir
```
and
```
byteorder
indexmap
serde_json
```

**Comment by torokati44** [comment]
Referencing https://github.com/mainmatter/cargo-autoinherit/issues/16 and https://github.com/mainmatter/cargo-autoinherit/issues/2 too.

File: core/build_playerglobal/Cargo.toml (hunk 1 above)
Comment (torokati44) [reply core/build_playerglobal/Cargo.toml:18]: This one is still there, no?

Comment (Dinnerbone) [reply core/build_playerglobal/Cargo.toml:18]: Oh shoot the lack of a space made me miss it

Comment (torokati44) [reply core/build_playerglobal/Cargo.toml]: Maybe the same happened with `downcast-rs`, `fnv`, and `walkdir` then? :)

**Comment by torokati44** [comment]
Also I think we could enable `serde/derive` globally, like `clap/drive`.

File: core/build_playerglobal/Cargo.toml (hunk 1 above)
Comment (torokati44) [reply core/build_playerglobal/Cargo.toml]: Scratch that, my mistake.

**Comment by torokati44** [comment]
Even ignoring my previous comments about this, I think `walkdir` is still declared four times.

**Comment by Dinnerbone** [comment]
> Also I think we could enable `serde/derive` globally, like `clap/derive`.

I kinda feel like derive shouldn't be workspace wide for either package tbh :sweat_smile: If we later get a crate that doesn't need derive, we'd have to go and undo this to allow that to build without needing derive.

But we aren't in that case today so I'll just do that for consistency

**Comment by torokati44** [comment]
Well then should we consider later making `clap/derive` not workspace-wide either? 🤔 Or is that definitely needed everywhere?

**Review by torokati44 — APPROVED** [review]
Awesome, thanks a lot!

(The last two commits are not in alphabetical order!! 😖🫠😵‍💫🫨😵😫😅😆😁😄)

**Comment by Dinnerbone** [comment]
> Well then should we consider later making `clap/derive` not workspace-wide either? 🤔 Or is that definitely needed everywhere?

It's needed everywhere right now. It's less likely that we'll use clap without derive, than serde without derive - so I'm happy for it to stay this way right now

**Review by sleepycatcoding — APPROVED** [review]

**Comment by sleepycatcoding** [comment]
The `walkdir` change is gonna create a local conflict on my save manager branch, but it is whatever. (since I also pulled it out into the workspace)

**Comment by torokati44** [comment]
I suspect it's gonna cause a few conflicts, if anything, due to neighboring line changes... But they should be easy to sort out.

**Pull request merged**